lib/client.js

```diff
@@ -128,15 +130,186 @@ function sendUncompressedMessage(options, message, logger) {
+        namespace = metricParams.namespace,
+        timestamp = metricParams.timestamp,
+        sampleRate = metricParams.sampleRate;
+
+    if (configHelper.areMetricTypesArgumentsValid(agg, aggFreq, tags, timestamp)) {
+        var putAgg = metricTypeConf ? configHelper.concatAggregations(metricTypeConf.agg, agg) : agg,
+            putTags = metricTypeConf ? merge(tags, metricTypeConf.tags) : tags,
+            putAggFreq = metricTypeConf ? (aggFreq || metricTypeConf.aggFreq) : aggFreq;
+
+        putRaw(self, name, value, {tags: putTags, agg: putAgg, aggFreq: putAggFreq, namespace: namespace, timestamp: timestamp, sampleRate: sampleRate}, isMetricAggregated, isStats);
+    } else if (self.logger) {
+        self.logger.warn('Metric not sent. Please review the following: aggregations, aggregation frequency, tags and timestamp.');
+    }
+}
+
+
+
 function sendFlushStats(self) {
     if (self.systemStats) {
         if (self.aggregatedBuffer.bufferSize > 0 && self.transport === 'api') {
-            self.put('buffer.flush_length', self.aggregatedBuffer.bufferSize, {agg: ['avg'], tags: {"buffer_type": 'aggregated'}});
+            putMetric(self, null, 'buffer.flush_length', self.aggregatedBuffer.bufferSize, null, null, {agg: ['avg'], tags: {"buffer_type": 'aggregated'}}, true);
         }
         if (self.nonAggregatedBuffer.bufferSize > 0) {
-            self.put('buffer.flush_length', self.nonAggregatedBuffer.bufferSize, {agg: ['avg'], tags: {"buffer_type": 'non-aggregated'}});
+            putMetric(self, null, 'buffer.flush_length', self.nonAggregatedBuffer.bufferSize, null, null, {agg: ['avg'], tags: {"buffer_type": 'non-aggregated'}}, true);
+        }
+    }
+}
+
+/**
+ * Logs all the metrics to the logger
+ *
+ * @param self A self client instance.
+
+ */
+function logMetrics(self) {
+    var stringToLogHeader = '';
+    var stringToLog = '';
+    var aggLog;
+    var aggFreqLog;
+
+    if (self.nonAggregatedBuffer.bufferSize > 0) {
+        self.logger.debug('Flushing metrics (non aggregated): ' + self.nonAggregatedBuffer.buffer);
+    }
+    if (self.aggregatedBuffer.bufferSize > 0) {
+        stringToLogHeader = 'Flushing metrics (aggregated): ';
+
+        for (aggLog in self.aggregatedBuffer) {
+            for (aggFreqLog in self.aggregatedBuffer[aggLog]) {
+                if (self.aggregatedBuffer[aggLog][aggFreqLog].buffer.length > 0) {
+                    if (stringToLog.length > 0) {
+                        stringToLog +='\n';
+                    }
+                    stringToLog += self.aggregatedBuffer[aggLog][aggFreqLog].buffer;
+                    self.aggregatedBuffer[aggLog][aggFreqLog].buffer = '';
+                }
+            }
+        }
+        self.logger.debug(stringToLogHeader + stringToLog);
+    }
+    if (self.nonAggregatedStatsBuffer.bufferSize > 0) {
+        self.logger.debug('Flushing metrics (non aggregated): ' + self.nonAggregatedStatsBuffer.buffer);
```

Comment:
Maybe change to `'Flushing stats metrics: ' + ...` as stats metrics will never be aggregated.